src/platform/user/profile/actions/index.js

```diff
@@ -25,37 +26,37 @@ export function profileLoadingFinished() {
   };
 }
 
-export function getProfile() {
-  return (dispatch) => {
-    return fetch(`${environment.API_URL}/v0/user`, {
+export function refreshProfile() {
+  return async (dispatch) => {
+    const response = await fetch(`${environment.API_URL}/v0/user`, {
       method: 'GET',
       headers: new Headers({
         Authorization: `Token token=${sessionStorage.userToken}`
       })
-    }).then(response => {
-      if (response.ok) return response.json();
+    });
+
+    if (!response.ok) {
       const error = new Error(response.statusText);
       error.status = response.status;
       throw error;
-    }).then(payload => {
-      const userData = payload.data.attributes.profile;
-      // sessionStorage coerces everything into String. this if-statement
-      // is to prevent the firstname being set to the string 'Null'
-      if (userData.first_name) {
-        sessionStorage.setItem('userFirstName', userData.first_name);
-      }
-      // Report out the current level of assurance for the user
-      recordEvent({ event: `login-loa-current-${userData.loa.current}` });
-      dispatch(updateProfileFields(payload));
+    }
+
+    const payload = await response.json();
+    dispatch(updateProfileFields(payload));
+    return payload;
+  };
+}
```

Comment:
I was a unsure of whether there should be any error handling here. With `restoreProfile` it makes sense to check for a `401` and take action if necessary, but I'm not sure if that's appropriate when doing a refresh.